invoices.go

```diff
@@ -209,9 +213,46 @@ func (i *InvoiceCollection) UnmarshalXML(d *xml.Decoder, start xml.StartElement)
 		Transactions:            v.ChargeInvoice.Transactions,
 		CreditPayments:          v.ChargeInvoice.CreditPayments,
 	}
+	var creditInvoices []Invoice
```

Comment:
Allocate the slice with the correct size (since it's known). Otherwise Go will allocate a bigger underlying array on each iteration of the loop.

```suggestion
	creditInvoices := make([]Invoice, len(v.CreditInvoices))
```

In the for loop swap `creditInvoices = append(creditInvoices, ...)` with `creditInvoices[i] = ...`

Reply:
done